### PR TITLE
make webclient a npm package

### PR DESCRIPTION
**Pull request by toloudis**
Final tweaks to publish the webclient as a npm package that other js projects can use.  Bring version number in sync with Agave release and python client (all in lockstep).
There may be some more tweaks upcoming, while testing the gh action to ensure that everything publishes correctly.
The main client code includes a change to allow better error handling when failing to connect.

### CODE REVIEW

File: webclient/src/agave.ts [hunk 1]
```diff
@@ -34,69 +35,77 @@ export class AgaveClient {
...
+      this.socket.onerror = (evt: Event) => {
+        console.log("error", evt);
+        reject(evt);
+      };
+    });
   }
 
   isReady(): boolean {
-    return this.socket.readyState === 1;
+    return this.socket!.readyState === 1;
```
Comment (ShrimpCryptid) [inline webclient/src/agave.ts:104]: Warning from linter for non-null assertion?

**Review by ShrimpCryptid — APPROVED** [review]
Looks good, added a few minor comments!

File: webclient/src/agave.ts [hunk 2]
```diff
@@ -34,69 +35,77 @@ export class AgaveClient {
...
+        resolve(this);
+      };
+
+      // TODO - handle this better
+      this.socket.onclose = (_ev: CloseEvent) => {
+        setTimeout(function () {
+          console.warn("connection failed. refresh to retry.");
+        }, 3000);
+      };
+
```
Comment (interim17) [inline webclient/src/agave.ts:73]: Would you want this code to have a try/catch that tries to reestablish connection before asking the user to refresh?

File: webclient/package.json [hunk 3]
```diff
@@ -26,22 +31,27 @@
...
     "@babel/register": "^7.21.0",
+    "@types/chai": "^4.3.5",
     "@types/dat.gui": "^0.7.10",
+    "@types/mocha": "^10.0.1",
     "@types/three": "^0.152.1",
     "@typescript-eslint/eslint-plugin": "^5.60.1",
     "@typescript-eslint/parser": "^5.60.1",
     "acorn": "^8.8.2",
     "babel-loader": "^9.1.2",
+    "chai": "^4.3.7",
```
[the remark this answers is not in the document]
Comment (toloudis) [reply webclient/package.json:42]: They often get out of sync especially with patch releases.  I selected "most current" at this time.

File: webclient/src/agave.ts (hunk 2 above)
Comment (toloudis) [reply webclient/src/agave.ts:73]: for now there aren't really any known use cases where this should happen.  This is really just some placeholder code (there's a TODO above). I'll simplify it even more.

File: webclient/package.json [hunk 4]
```diff
@@ -1,20 +1,25 @@
 {
-  "name": "agave-webclient",
-  "version": "1.0.0",
+  "name": "@aics/agave-webclient",
+  "version": "1.5.0",
```
Comment (ShrimpCryptid) [inline webclient/package.json:3]: Is it possible to have this auto-update with new release versions, or does this have to be manually updated?

File: webclient/src/commandbuffer.ts [hunk 5]
```diff
@@ -105,6 +105,7 @@ export class CommandBuffer {
 
       const command = this.prebuffer[i];
       const commandCode = command[0] as string;
+      //console.log("AGAVE." + commandCode, command.slice(1));
```
Comment (ShrimpCryptid) [inline webclient/src/commandbuffer.ts:108]: Should this be removed?

**Review by ShrimpCryptid — DISMISSED** [review]
Looks good! Left a few notes :)

File: webclient/package.json (hunk 4 above)
Comment (toloudis) [reply webclient/package.json:3]: there's a version bump script (tbump.toml) that goes to all the files that have these version number in them and updates all of them

File: webclient/src/commandbuffer.ts (hunk 5 above)
Comment (toloudis) [reply webclient/src/commandbuffer.ts:108]: sure - I'll remove it

**Pull request merged**